mii/grpc_related/modelresponse_server.py

```diff
@@ -190,41 +200,78 @@ def choose_stub(self, call_count):
     def intercept_service(self, continuation, handler_call_details):
         next_handler = continuation(handler_call_details)
         assert next_handler.unary_unary is not None
+        #USE KWARGS LIKE THEY ARE USED TO MAKE SESSIONS TO GET THE DEPLOYMENT NAME TO HASH THE COUNTERS/STUBS
 
         def invoke_intercept_method(request_proto, context):
             method_name = _get_grpc_method_name(handler_call_details.method)
+            kwargs = unpack_proto_query_kwargs(request_proto.query_kwargs)
```

Comment:
I don't think this is required. I believe we can grab the deployment name from kwargs here: https://github.com/microsoft/DeepSpeed-MII/blob/ce80314c0d086485b1559663102c48d2c24cb2eb/mii/client.py#L125

The `query_kwargs` is intended to be passed to the pipeline object - so using it to also capture the deployment name breaks that assumption.